### ISSUE TITLE
Cannot read property 'filter' of undefined

### ISSUE THREAD

**Issue by sibelius** [Automattic/mongoose#8128]
**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
it is breaking with the following error:

```jsx
TypeError: Cannot read property 'filter' of undefined

      at node_modules/mongoose/lib/model.js:3251:32
      at parallelLimit (node_modules/mongoose/lib/helpers/parallelLimit.js:15:12)
      at Function.Object.<anonymous>.Model.$__insertMany (node_modules/mongoose/lib/model.js:3245:3)
      at process.nextTick (node_modules/kareem/index.js:369:33)
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Everything worked well on `5.6.11` but it broke on `5.6.12`

**What is the expected behavior?**
it should not break

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
mongoose: 5.6.12
node: 10.15.0
mongo: v4.0.5

**Comment by nadr0** [comment]
Happened to me too. I rolled out the version with 5.6.12 and it reported the same error. I have to roll back to 5.6.11 and things are back to normal.

**Comment by vkarpov15** [comment]
@sibelius not a very helpful bug report - please provide code samples next time. Stack traces are typically not very helpful :-1:

**Comment by sibelius** [comment]
the error happens when you are saving many documents at once

the error is here

https://github.com/Automattic/mongoose/blob/master/lib/model.js#L3251

const docAttributes = docs.filter(function(doc) {
      return doc != null;
    });

docs is null in `parallelLimit`

maybe a regression from here

https://github.com/Automattic/mongoose/commit/03dfc6a90295d55203e3d93bbe5254c8d60451d7

**Comment by sibelius** [comment]
this could be the cause of the bug

https://github.com/Automattic/mongoose/commit/03dfc6a90295d55203e3d93bbe5254c8d60451d7#diff-af13acd1e8d4818a7336e3dd68eb5efbR15

if `fns.length === 0)`, it won't have either `error` nor `docs`

**Comment by nadr0** [comment]
@vkarpov15 

I cannot share my exact snippet of code that failed but it failed in this way.

```
// inside some async function ...
try {
  const arr = [];
  const result = await myModel.insertMany(arr);
} catch (e) {
  console.log(e);
}
```
I tried to do `insertMany` (https://mongoosejs.com/docs/api.html#model_Model.insertMany) with an empty array and I got the same error @sibelius got.

**Comment by vkarpov15** [comment]
Looks like @sibelius fixed this in his latest #8130. @nadr0 please upgrade to mongoose 5.6.13 for the fix.
